File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char,int>count;
        int start=0;
        int end=0;
        int res=0;
        while(end<s.size()){
            count[s[end]]++;
            while(count[s[end]]>1){
                count[s[start]]--;
                start++;
                }
            res=max(res,end-start+1);
            end++;
        }
        return res;
            
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int start=0;
        int end=0;
        int maxlen=Math.min(s.length(),1);
        Set<Character> seen=new HashSet<>();
        while(end<s.length())
        {
           char c=s.charAt(end);
           while(seen.contains(c))
           {
               seen.remove(s.charAt(start));
               start+=1;
           }
            seen.add(c);
            int windowsize=end-start+1;
            maxlen=Math.max(maxlen,windowsize);
            end+=1;
        }
    return maxlen;
    }
    
}